src/java/TypesUtils.cpp

```diff
@@ -342,6 +342,7 @@ static bool isJavaMap(JNIEnv * env, jobject obj)
 
 static Object * hashmapJavaToObjectConverter(JNIEnv * env, jobject obj)
 {
+	env->PushLocalFrame(20);
```

Comment:
The local object references are usually cleaned up when control is returned to the JVM. However, during the htmlRendering operation, numerous calls are made through the JNI to gather the data necessary for rendering. Control is not returned to the JVM until all the callback methods have been executed. So much data is returned from the callbacks that he local object reference table fills up (512 entries) and causes app termination. This just releases the object references as soon as they are no longer required.